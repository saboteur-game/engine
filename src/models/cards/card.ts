import { generateId, Pojo } from "../../utils";
import CardParameters from "./card-parameters";

export enum Status {
  unused = "unused",
  played = "played",
  discarded = "discarded",
}

class Card {
  id: string;
  status: Status;

  constructor() {
    this.id = generateId();
    this.status = Status.unused;
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  play(parameters: CardParameters): Card {
    return this;
  }

  setDiscarded(): void {
    // TODO: Check if the state was unused and throw if it was anything else
    this.status = Status.discarded;
  }

  setPlayed(): void {
    // TODO: Check if the state was unused or played and throw if it was anything else
    this.status = Status.played;
  }

  toJS(): Pojo {
    return {
      id: this.id,
      status: this.status,
    };
  }
}

export default Card;
