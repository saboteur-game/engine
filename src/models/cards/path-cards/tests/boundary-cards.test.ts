import { Sides } from "../../card";
import { StartPathCard, GoldFinishPathCard, RockFinishPathCard } from "..";

describe("Boundary Cards", () => {
  describe("StartPathCard", () => {
    it("can be visualized", () => {
      expect(new StartPathCard().visualize()).toMatchSnapshot();
    });
  });

  describe("GoldFinishPathCard", () => {
    it("can be visualized", () => {
      expect(new GoldFinishPathCard().visualize()).toMatchSnapshot();
    });
  });

  describe("GoldFinishPathCard", () => {
    it("can be visualized right side up", () => {
      expect(
        new RockFinishPathCard([Sides.top, Sides.right], false).visualize()
      ).toMatchSnapshot();
    });

    it("can be visualized upside down", () => {
      expect(
        new RockFinishPathCard([Sides.top, Sides.right], true).visualize()
      ).toMatchSnapshot();
    });
  });
});
