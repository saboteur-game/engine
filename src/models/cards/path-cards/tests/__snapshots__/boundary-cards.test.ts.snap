// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boundary Cards GoldFinishPathCard can be visualized 1`] = `
"██  ██
  !!  
██  ██"
`;

exports[`Boundary Cards GoldFinishPathCard can be visualized right side up 1`] = `
"██  ██
██xx  
██████"
`;

exports[`Boundary Cards GoldFinishPathCard can be visualized upside down 1`] = `
"██████
  xx██
██  ██"
`;

exports[`Boundary Cards StartPathCard can be visualized 1`] = `
"██  ██
  []  
██  ██"
`;
